### DISCUSSION TITLE
Dependency on io.quarkus no longer exists. io.quarkus:quarkus-test-security-jwt not found in catalog

### DISCUSSION THREAD

**Discussion by nkolatsis** [quarkusio/quarkus#41276] [Q&A] [answered]
I'm new to quarkus and I'm following the Quarkus + React book from Marc Nuri San Felix. The book is only a year old and uses Quarkus 2.10 so I would have thought that legacy dependencies would stick around and be working for far longer than a year. How would you proceed from this issue?

This is the error I am getting is this:
```

nick@mint-T490s:~/code/Full-Stack-Quarkus-and-React/active-project$ ./mvnw quarkus:add-extension -Dextensions=io.quarkus:quarkus-test-security-jwt
[INFO] Scanning for projects...
[INFO] 
[INFO] -------------------< com.example.fullstack:reactive >-------------------
[INFO] Building reactive 1.0.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- quarkus-maven-plugin:2.10.2.Final:add-extension (default-cli) @ reactive ---
[INFO] [ERROR] ❗  Nothing installed because keyword(s) 'io.quarkus:quarkus-test-security-jwt' were not matched in the catalog.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.198 s
[INFO] Finished at: 2024-06-18T11:44:52+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:2.10.2.Final:add-extension (default-cli) on project reactive: Unable to update the pom.xml file: Unable to add extensions -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

[a comment by an automated account is not shown]

**Comment by sberyozkin** [comment]
Hi @nkolatsis, `quarkus-test-security-jwt` is not an extension, it is a test support module. Is it how it is shown to be added in Marc's [book](https://www.amazon.co.uk/Full-Stack-Quarkus-React-Hands) ?

**Reply by sberyozkin** [reply]
Let me CC to Marc, @manusa

**Reply by nkolatsis** [reply]
Hi @sberyozkin @manusa

I have attached a photo of the bottom of page 86. When searching for `quarkus-test-security-jwt` in this repo ([link](https://github.com/search?q=repo%3Aquarkusio%2Fquarkus%20quarkus-test-security-jwt&type=code)), I found many references that show it as a dependency including this bit of code.

```

            <dependency>
                <groupId>io.quarkus</groupId>
                <artifactId>quarkus-test-security-jwt</artifactId>
                <version>${project.version}</version>
            </dependency>
```

![WhatsApp Image 2024-06-18 at 12 53 47](https://github.com/quarkusio/quarkus/assets/13288906/f5119049-25f6-4ed2-b5a1-a59da88a029e)

**Reply by manusa** [reply]
This is the official guide where the module is explained: https://quarkus.io/guides/security-jwt#integration-testing-security-annotation

**Comment by manusa** [comment, marked as answer]
Hi @nkolatsis 

Thanks for checking out the book.
Unfortunately there is no second edition yet so some of the code and commands shown in the book have become obsolete.

You can find an updated code-base here: https://github.com/manusa/packt-fullstack-quarkus-react/tree/feat/quarkus-3/spike-v3

You should be able to manually add the dependency to your pom.xml file:
```xml
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-test-security-jwt</artifactId>
      <scope>test</scope>
    </dependency>
```

See https://github.com/manusa/packt-fullstack-quarkus-react/blob/946a4400dfb5ceda3d286a7d6ebe402ec72875cd/spike-v3/pom.xml#L75-L79
